packages/core/contracts/proxy-scripts/lsp-broker/LspUniswapV2Broker.sol

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "@uniswap/lib/contracts/libraries/TransferHelper.sol";
+import "@uniswap/v2-periphery/contracts/interfaces/IUniswapV2Router01.sol";
+
+import "../../common/implementation/FixedPoint.sol";
+
+import "../../financial-templates/long-short-pair/LongShortPair.sol";
+
+/**
+ * @title LspUniswapV2Broker
+ * @notice Helper contract to facilitate batched LSP and UniswapV2 transactions, including Mint+Sell and Mint+LP.
+ */
+contract LspUniswapV2Broker {
+    using FixedPoint for FixedPoint.Unsigned;
+
+    /**
+     * @notice Mint long and short tokens and deposit them all into a UniV2 Pool.
+     */
+    function atomicMintAddLiquidity() public {
+        /* TODO */
+    }
+
+    /**
+     * @notice Mint long and short tokens and convert all of one side into the other.
+     * @param tradingAsEOA If True, caller has balance of collateral and expects to receive back all long/short tokens.
+     * @param sellLong If True, converts all long tokens into short, else the opposite.
+     * @param longShortPair LSP contract address to mint position on.
+     * @param router Contract to call to exchange long and short tokens.
+     * @param collateralToMintWith Amount of collateral to deposit and borrow long and short tokens against.
+     * @param swapPath `Router.swapExactTokensForTokens` param: path with which to swap token to sell for the other.
+     * @param deadline `Router.swapExactTokensForTokens` param: time before transaction must be mined.
+     */
+    function atomicMintSellOneSide(
+        bool tradingAsEOA,
+        bool sellLong,
+        LongShortPair longShortPair,
+        IUniswapV2Router01 router, /* TODO: Should we allow `router` to be any exchange, such as a Matcha multihop? */
+        uint256 collateralToMintWith,
+        address[] memory swapPath,
+        uint256 deadline
+    ) public {
+        require(address(longShortPair) != address(0), "Invalid long short pair");
+        require(address(router) != address(0), "Invalid router");
+        require(collateralToMintWith != 0, "Collateral to mint with");
+
+        // 0) Pull collateral from caller if necessary and approve LSP to spend it.
+        if (tradingAsEOA)
+            TransferHelper.safeTransferFrom(
+                address(longShortPair.collateralToken()),
+                msg.sender,
+                address(this),
+                collateralToMintWith
+            );
+
+        TransferHelper.safeApprove(
+            address(longShortPair.collateralToken()),
+            address(longShortPair),
+            collateralToMintWith
+        );
```

Comment:
nit: why do we use uniswap's safe transfer libraries rather than OZ (which is what we generally use elsewhere)? Is there an advantage to theirs?

Reply:
OZ's `SafeERC20` works by overriding some of the behavior of `IERC20` or `ExpandedIERC20` (depending on the import) to ensure that token actions will throw on failure, rather than returning false. In my work previously with the DSProxy scripts I found the uniswap method `TransferHelper` to use _less_ gas and bytecode than the OZ implementation as you don't need to instantiate an instance of the contract you are calling into. Granted, you could do the same thing with the OZ libs by declaring it inline then calling methods on it but I found that even then the uniswap implementation used less. 

The pattern that I've gone for with these scripts is any code within the `proxy-scripts` directory (which includes things to help using a DSProxy or code like this file) uses the Uniswap helper and the rest of the repo uses the OZ implementation. 

I'm happy to refactor these to all use the OZ implementation but we might hit bytecode issues again in these scripts if they are on the margin.

Reply:
I'm in favor of keeping Uniswap's transfer helper for consistency within this directory